### PR TITLE
Moved term constants and helper functions to shared/terms

### PR DESCRIPTION
**Pull request by airenzp**
## Description

Moved term type enumerators, mappings and helper functions to shared/terms. Updated Imports. Fixed handler imports using #shared. Used constants in typeGroups definition. I also added added the groupTypes definition in case we want to add it, if not we can remove it. Please check.

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

**Comment by gavrielm** [comment]
Thanks for the refactoring. All changes seem to work well.

But I do not think the `groupTypes` definition is worth adding right now. If we need to add a new term type, we would need to add it to `TermTypes`, `typeGroup`, `groupTypes`, and `nonDictTypes` in `client/shared/terms.js`. That can be difficult to maintain. Ideally, we could have a single object to store all term types and then work off of that object. Not sure if that is feasible right now, but something to possibly work towards.

**Review by xzhou82 — APPROVED** [review]
all CI pass
in termdb.matrix.js, `sampleFilterSet` is still used by geneExp but it should not

**Comment by airenzp** [comment]
> all CI pass in termdb.matrix.js, `sampleFilterSet` is still used by geneExp but it should not

fixed now

File: server/src/termdb.matrix.js [hunk 1]
```diff
@@ -110,7 +110,8 @@ async function getSampleData(q) {
 		return { samples, refs: { byTermId, bySampleId: {} } }
 	}
 
-	const sampleFilterSet = await mayGetSampleFilterSet(q, nonDictTerms) // conditionally returns a set of sample ids
+	const sampleFilterSet = await mayGetSampleFilterSet4snplst(q, nonDictTerms) // conditionally returns a set of sample ids, FIXME *only* for snplst and snplocus data download in supported ds, not for anything else. TODO remove this bad quick fix
```
Comment (airenzp) [inline server/src/termdb.matrix.js:113]: can we move this line to the if case handling snp cases then?

**Comment by xzhou82** [comment]
then geneExp query breaks

**Comment by airenzp** [comment]
please send the example

**Comment by airenzp** [comment]
Moved mayGetSampleFilterSet4snplst to snp cases

**Review by gavrielm — APPROVED** [review]

**Pull request merged**